### PR TITLE
Improve docker build

### PR DESCRIPTION
**Pull request by bnemanich**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/master/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/master/doc/)?
- [ ] Did you write any [tests](https://github.com/open-ce/open-ce/blob/master/tests/) to validate this change?  

## Description

This will add the capability to use anonymous volumes for temporary build artifacts when using the `--docker_build` option with the build_env script. This can help with docker containers that have limited disk space.

Thanks to @dllehr81 for the advice!

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

File: open-ce/utils.py [hunk 1]
```diff
@@ -82,6 +82,30 @@ class Argument(Enum):
                                         default=DEFAULT_BUILD_TYPES,
                                         help='Comma delimited list of build types, such as "cpu" or "cuda".'))
 
+    DOCKER_BUILD = (lambda parser: parser.add_argument(
+                                        '--docker_build',
+                                        action='store_true',
+                                        help="Perform a build within a docker container. "
+                                             "NOTE: When the --docker_build flag is used, all arguments with paths "
+                                             "should be relative to the directory containing open-ce. Only files "
+                                             "within the open-ce directory and local_files will be visible at "
```
Comment (dllehr81) [inline open-ce/utils.py:91]: I think this should be "Only files and folders within the current working directory will be visible"  unless you're forcing them to be run at a specific level.

File: open-ce/utils.py [hunk 2]
```diff
@@ -82,6 +82,30 @@ class Argument(Enum):
...
+                                        '--use_local_scratch',
+                                        action='store_true',
+                                        help="When performing a build within a container, use local scratch space "
+                                             "instead of disk space within the container. This can be useful if the "
+                                             "container has limited disk space. If `local_scratch_folder` is not "
+                                             "set, scratch space will be taken from a temporary folder, which will "
+                                             "be deleted after the job has completed."))
+
+    LOCAL_SCRATCH_FOLDER = (lambda parser: parser.add_argument(
+                                        '--local_scratch_folder',
```
Comment (dllehr81) [inline open-ce/utils.py:104]: Minor nit, but you reference directories above, and now folders here.  Is there a difference in the terminology?

File: open-ce/utils.py [hunk 3]
```diff
@@ -82,6 +82,30 @@ class Argument(Enum):
...
+                                        '--docker_build',
+                                        action='store_true',
+                                        help="Perform a build within a docker container. "
+                                             "NOTE: When the --docker_build flag is used, all arguments with paths "
+                                             "should be relative to the directory containing open-ce. Only files "
+                                             "within the open-ce directory and local_files will be visible at "
+                                             "build time."))
+
+    USE_LOCAL_SCRATCH = (lambda parser: parser.add_argument(
+                                        '--use_local_scratch',
```
Comment (dllehr81) [inline open-ce/utils.py:95]: May be worth adding what's being pulled locally, i.e. the `.cache` and `conda-bld` directories...if it gets too wordy I wouldn't worry about it.

File: open-ce/docker_build.py [hunk 4]
```diff
@@ -21,6 +24,15 @@
 
 DOCKER_TOOL = "docker"
 
+def make_parser():
+    ''' Parser for input arguments '''
+    arguments = [utils.Argument.DOCKER_BUILD]
+    parser = utils.make_parser(arguments,
+                               description='Run Open-CE tools within a container',
+                               formatter_class=argparse.ArgumentDefaultsHelpFormatter)
```
Comment (bethune-bryant) [inline open-ce/docker_build.py:32]: `make_parser` defaults to the formatter from utils. Then you wouldn't have to import argparse.

```suggestion
    parser = utils.make_parser(arguments, description='Run Open-CE tools within a container')
```

**Comment by bethune-bryant** [comment]
This is a lot cleaner now!

File: open-ce/docker_build.py [hunk 5]
```diff
@@ -9,6 +9,9 @@
 import os
 import datetime
 import platform
+import argparse
```
Comment (bnemanich) [inline open-ce/docker_build.py:12]:
```suggestion
```

**Review by bethune-bryant — APPROVED** [review]
